include/exadg/vector_tools/arbitrary_solution_transfer.h

```diff
@@ -0,0 +1,167 @@
+         dealii::Mapping<dim> const &    mapping_src_in)
+  {
+    dof_handler_dst = &dof_handler_dst_in;
+    dof_handler_src = &dof_handler_src_in;
+
+    rpe.reinit(collect_mapped_support_points(dof_handler_dst_in, mapping_dst_in),
+               dof_handler_src_in.get_triangulation(),
+               mapping_src_in);
+  }
+
+  /**
+   * Transfer the solution between source and target triangulations, that were set with reinit.
+   * At support points which are not overlapping the source triangulation, corresponding dof entries
+   * are set to 0.
+   *
+   * @param[in] dst Target Dof Vector.
+   * @param[in] src Source Dof Vector.
+   */
+  template<int n_components, typename VectorType1, typename VectorType2>
+  void
+  transfer_solution(VectorType1 &                                               dst,
+                    VectorType2 const &                                         src,
+                    dealii::VectorTools::EvaluationFlags::EvaluationFlags const flags =
+                      dealii::VectorTools::EvaluationFlags::avg) const
+  {
+    src.update_ghost_values();
+    const auto values =
+      dealii::VectorTools::point_values<n_components>(rpe, *dof_handler_src, src, flags);
+    src.zero_out_ghost_values();
+
+    fill_dof_vector_with_values<n_components>(dst, *dof_handler_dst, values);
+  }
+
+private:
+  std::vector<dealii::Point<dim>>
+  collect_mapped_support_points(dealii::DoFHandler<dim> const & dof_handler,
+                                dealii::Mapping<dim> const &    mapping) const
+  {
+    std::vector<dealii::Point<dim>> global_mapped_support_points;
+    std::vector<dealii::Point<dim>> mapped_support_points;
+
+    for(auto const & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      mapped_support_points.resize(cell->get_fe().get_generalized_support_points().size());
+
+      unsigned int idx = 0;
+      for(auto const & sp : cell->get_fe().get_generalized_support_points())
+        mapped_support_points[idx++] = mapping.transform_unit_to_real_cell(cell, sp);
+
+      global_mapped_support_points.insert(global_mapped_support_points.end(),
+                                          mapped_support_points.begin(),
+                                          mapped_support_points.end());
+    }
+
+    return global_mapped_support_points;
+  }
+
+  template<int n_components, typename VectorType, typename ValueType>
+  void
+  fill_dof_vector_with_values(VectorType &                    dst,
+                              dealii::DoFHandler<dim> const & dof_handler,
+                              std::vector<ValueType> const &  values) const
+  {
+    auto ptr = values.begin();
+    for(const auto & cell : dof_handler.active_cell_iterators())
+    {
+      if(cell->is_locally_owned() == false)
+        continue;
+
+      unsigned int n_dofs_per_component = cell->get_fe().get_generalized_support_points().size();
```

Comment:
unsigned int const?

Reply:
`n_nodes` could be a more intuitive name

Reply:
The naming should be chosen as above. So if we go with "nodes" I am fine with this, if we go with "support_points" I will change it to n_support_points

Reply:
agree